### ISSUE TITLE
at-rule-no-unknown errors

### ISSUE THREAD

**Issue by neverfox** [kristerkari/stylelint-scss#86]
I get at-rule-no-unknown errors for things like @extend, @mixin etc. with this plugin installed. I could be mistaken, but I assumed this plugin would cover those cases, but right now I have to use a custom ignore rule:

```
{
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-css-modules"
  ],
  "plugins": [
    "stylelint-scss"
  ],
  "rules": {
    "at-rule-no-unknown": [ true, {
      ignoreAtRules: ['extend', 'at-root', 'debug', 'warn', 'error', 'if', 'else', 'for', 'each', 'while', 'mixin', 'include', 'content', 'return', 'function']
    }]
  }
}
```

**Comment by dryoma** [comment]
The list of *known* rules is defined in [stylelint's code](https://github.com/stylelint/stylelint/blob/master/src/reference/keywordSets.js#L487). This project loads stylelint as a dependency, thus we can use its code in our code, but can not alter the rules from a standalone stylelint users define in their configs. ... or can we @davidtheclark ? Is it possible to give a plugin set the ability to redefine things like objects in keywordSets?

Alternatively something like `stylelint-scss/at-rule-no-unknown` can be used instead of the core rule; it would be the same except it would utilize our extended set of known rules. The problem would be we'll also have to keep up with the currently applicable standart atrules, so the first option I think is preferrable.

**Comment by eriklharper** [comment]
@neverfox I'm in the same boat, but what I realized is that because `stylelint-scss` is a stylelint `plugin`, it doesn't automatically enable any rule for you because [that's how plugins work](http://stylelint.io/user-guide/configuration/#plugins).

What we _really_ need to make the linter "scss aware" is a scss lint config that we can tell stylelint to `extend` instead, the same way you're extending `stylelint-config-css-modules` above.

I did a quick search online for "stylelint scss config" but it isn't turning up anything, perhaps this is a perfect moment for me to attempt to write one myself!

**Comment by eriklharper** [comment]
Here we go, there actually is one: https://github.com/bjankord/stylelint-config-sass-guidelines/blob/master/index.js.  See how he's using `stylelint-scss` as a plugin and then implementing `rules` on top of that plugin to make it scss aware.

**Comment by neverfox** [comment]
@eriklharper Cool. Makes sense. Thanks for doing the leg work.

**Comment by davidtheclark** [comment]
Another possibility would be to write a new rule for `stylelint-scss` that *wraps* `at-rule-no-unknown` but automatically ignores Sass's special at-rules. Something like this:

```js
var sassAtRules = [
  'extend', 'at-root', 'debug', 
  'warn', 'error', 'if', 'else', 
  'for', 'each', 'while', 'mixin', 
  'include', 'content', 'return', 'function'
];

function atRuleNoUnknownSassy(on, options) {
  return function (root, result) {
    const defaultedOptions = Object.assign({}, options, {
      ignoreAtRules: sassAtRules.concat(options.ignoreAtRules || [])
    });

    stylelint.rules['at-rule-no-unknown'](on, defaultedOptions)(root, result)''
  };
};
```

This solution means your rule is easy to maintain, without duplicating code and at-rule lists in stylelint.

This ability to manually invoke stylelint rules *within plugin rules* opens a lot of doors!

**Comment by dryoma** [comment]
@davidtheclark awesome! Will give it a shot a bit later.

**Comment by jeddy3** [comment]
> Another possibility would be to write a new rule for stylelint-scss that wraps at-rule-no-unknown

@dryoma FYI, this is documented in [this section](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md#stylelintrules) of the developer guide. It's also worth keeping an eye on https://github.com/stylelint/stylelint/pull/2173 in case it impacts on how wrapping rules might work going forward.

**Comment by kristerkari** [comment]
This is implemented in version 1.5

**Comment by iwanttobealight** [comment]
> I get at-rule-no-unknown errors for things like @extend, @mixin etc. with this plugin installed. I could be mistaken, but I assumed this plugin would cover those cases, but right now I have to use a custom ignore rule:
> 
> ```
> {
>   "extends": [
>     "stylelint-config-standard",
>     "stylelint-config-css-modules"
>   ],
>   "plugins": [
>     "stylelint-scss"
>   ],
>   "rules": {
>     "at-rule-no-unknown": [ true, {
>       ignoreAtRules: ['extend', 'at-root', 'debug', 'warn', 'error', 'if', 'else', 'for', 'each', 'while', 'mixin', 'include', 'content', 'return', 'function']
>     }]
>   }
> }
> ```

thank you! :octocat:
